shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -54,11 +54,45 @@ - (UIAccessibilityContrast)accessibilityContrast;
 #endif
 
 @interface FlutterViewController (Tests)
+- (void)surfaceUpdated:(BOOL)appeared;
 - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences;
 @end
 
 @implementation FlutterViewControllerTest
 
+- (void)testViewDidDisappearDoesntPauseEngine {
+  id engine = OCMClassMock([FlutterEngine class]);
+  id lifecycleChannel = OCMClassMock([FlutterBasicMessageChannel class]);
+  OCMStub([engine lifecycleChannel]).andReturn(lifecycleChannel);
+  FlutterViewController* viewControllerA = [[FlutterViewController alloc] initWithEngine:engine
+                                                                                 nibName:nil
+                                                                                  bundle:nil];
+  FlutterViewController* viewControllerB = [[FlutterViewController alloc] initWithEngine:engine
+                                                                                 nibName:nil
+                                                                                  bundle:nil];
+  id viewControllerMock = OCMPartialMock(viewControllerA);
+  OCMStub([viewControllerMock surfaceUpdated:NO]);
+  OCMStub([engine viewController]).andReturn(viewControllerB);
+  [viewControllerA viewDidDisappear:NO];
+  OCMReject([lifecycleChannel sendMessage:@"AppLifecycleState.paused"]);
+  OCMReject([viewControllerMock surfaceUpdated:[OCMArg any]]);
+}
+
+- (void)testViewDidDisappearDoesPauseEngine {
```

Comment:
Can we tweak the names a bit? Sounds like we're testing for 2 different outcomes with the same conditions :)